Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.h

```diff
@@ -136,8 +136,9 @@ class RemoteSerializedImageBufferProxy : public WebCore::SerializedImageBuffer {
 
     WebCore::ImageBuffer::Parameters m_parameters;
     WebCore::ImageBufferBackend::Info m_info;
-    WebCore::RenderingResourceIdentifier m_renderingResourceIdentifier;
+    RemoteSerializedImageBufferReferenceTracker m_renderingResourceReferenceTracker;
```

Comment:
In general, only mutable objects need real reference trackers.
An immutable object only has two types of references: the read reference being used to read it, and then write reference created from the read references when the destroying write happens.

In particular, SerializedImageBuffer does not need reference trackers at all.
There's no operation whatsoever that the holder can do to it, apart from destroying it by losing the reference or destroying it by moving it to ImageBuffer. Thus by definition code that would add a reference tracker does not make sense.

Reply:
Hmm
I'm using reference trackers to enforce ordering of writes/adds (which works in this case).
I did notice that the API is more geared towards reads/writes.